src/modules/NavContainer/NavContainer.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import NavBar from '../../NavBar/NavBar';
+import SideMenu from '../../SideMenu/SideMenu';
+
+const GET_UPCOMING_EVENTS_URL =
+	'https://2sif0durcj.execute-api.eu-west-1.amazonaws.com/dev/get-upcoming-events';
+
+export default class NavContainer extends Component {
+	state = {
+		drawerMenu: false,
+		nextEvent: null,
+	};
+
+	componentDidMount() {
+		this.fetchUpcomingEvent.call(this);
+	}
+
+	async fetchUpcomingEvent() {
+		try {
+			let url = GET_UPCOMING_EVENTS_URL;
+			if (process.env.REACT_APP_STAGE === 'dev') {
+				url = '/mock/getUpcomingEvents.json';
+			}
+			const response = await fetch(url).then(data => data.json());
+			const nextEvent = response[0];
+			if (nextEvent) {
+				this.setState({ nextEvent });
+			}
+		} catch (err) {
+			// fail silently
+		}
+	}
```

Comment:
This `fetchUpcomingEvent` could also be a util function since you use it in multiple places. Then the code in the components can be shortened :-)